### PR TITLE
Blog - How to validate and clean your YAML files using Kubeval and ValidKube

### PR DESCRIPTION
**Pull request by msali123**
Please have a go through of code area's having the filenames.

### CODE REVIEW

**Comment by msali123** [comment]
@jalletto

**Comment by msali123** [comment]
Removed conflict.

**Comment by msali123** [comment]
@jalletto  can you please review

**Comment by msali123** [comment]
@jalletto can you please review

**Comment by jalletto** [comment]
Commit 0eb0e8bb978878b1ba3fc59bcdddcde639c91b05 deployed to https://62d9840e16cbef00616c49ae--ephemeral-narwhal-3ac3b2.netlify.app/

File: blog/_posts/2022-08-21-validate-&-clean-yaml.md [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+
+Once the download is complete, you can confirm Kubeval has been installed successfully by using the following command:
+
+~~~{.bash caption=">_"}
+kubeval --version
+~~~
+
+You will get the following output message that shows the Kubeval version:
+
+~~~{.bash caption=">_"}
```
Comment (jalletto) [inline blog/_posts/2022-08-21-validate-&-clean-yaml.md:49]:
```suggestion
~~~{.bash caption="output"}
```

File: blog/_posts/2022-08-21-validate-&-clean-yaml.md [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+
+The above `ClusterRole` gives the user the right to get, watch and list secrets. Use the following command to validate the file:
+
+~~~{.bash caption=">_"}
+kubeval secret-reader.yaml
+~~~
+
+You will get the following output:
+
+~~~{.bash caption=">_"}
```
Comment (jalletto) [inline blog/_posts/2022-08-21-validate-&-clean-yaml.md:85]:
```suggestion
~~~{.bash caption="output"}
```

File: blog/_posts/2022-08-21-validate-&-clean-yaml.md [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+
+Use the following command to validate the above YAML file.
+
+~~~{.bash caption=">_"}
+kubeval service-ingress.yml
+~~~
+
+You will get the following output that shows that the missing key error has been detected:
+
+~~~{.bash caption=">_"}
```
Comment (jalletto) [inline blog/_posts/2022-08-21-validate-&-clean-yaml.md:118]:
```suggestion
~~~{.bash caption="output"}
```

File: blog/_posts/2022-08-21-validate-&-clean-yaml.md [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+    protocol: TCP
+    targetPort: 80
+  selector:    
+    app: nginx
+  type: NodePort
+~~~
+
+If you scan the above contents using Kubeval again you will get the following output:
+
+~~~{.bash caption=">_"}
```
Comment (jalletto) [inline blog/_posts/2022-08-21-validate-&-clean-yaml.md:148]:
```suggestion
~~~{.bash caption="output"}
```

File: blog/_posts/2022-08-21-validate-&-clean-yaml.md [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+
+Use the following command to get the validation results in TAP format:
+
+~~~{.bash caption=">_"}
+kubeval secret-reader.yaml -o tap
+~~~
+
+You will get the following output:
+
+~~~{.bash caption=">_"}
```
Comment (jalletto) [inline blog/_posts/2022-08-21-validate-&-clean-yaml.md:184]:
```suggestion
~~~{.bash caption="output"}
```

File: blog/_posts/2022-08-21-validate-&-clean-yaml.md [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+
+The `--ignore-missing-schemas` flag should be used when validating resources using [Custom Resource Definitions(CRDs)](Custom Resources | Kubernetes). Currently, Kubeval relies on schemas generated from the Kubernetes API. This means it's not possible to validate resources using CRDs. Kubeval currently supports schemas created by the Kubernetes API only.
+
+~~~{.bash caption=">_"}
+kubeval --ignore-missing-schemas service-ingress.yml
+~~~
+
+You will get the following output:
+
+~~~{.bash caption=">_"}
```
Comment (jalletto) [inline blog/_posts/2022-08-21-validate-&-clean-yaml.md:213]:
```suggestion
~~~{.bash caption="output"}
```

**Review by jalletto — CHANGES_REQUESTED** [review]

**Review by jalletto — APPROVED** [review]

**Pull request merged**